### PR TITLE
[Feature] adds the sub tag

### PR DESCRIPTION
**Pull request by grahamjenson**
Adds the ability for the `!sub` tag which replaces a strings values with references

### CODE REVIEW

File: lib/geoengineer/gps/yaml_tag.rb [hunk 1]
```diff
@@ -83,6 +83,32 @@ def references
...
+      result = finder.dereference!(query, { auto_load: false })
+      result_value.gsub!("{{#{query}}}", result)
+    end
+    result_value.to_json
+  end
+
+  def references
+    all_refs = []
+    all_queries.each do |query|
+      components = query.match(GeoEngineer::GPS::Finder::NODE_REFERENCE_SYNTAX)
```
Comment (brad-larson) [inline lib/geoengineer/gps/yaml_tag.rb]: Shouldn't this be `CONSTANT_REFERENCE_SYNTAX`?

File: lib/geoengineer/gps/yaml_tag.rb [hunk 2]
```diff
@@ -83,6 +83,32 @@ def references
...
+    all_queries.each do |query|
+      result = finder.dereference!(query, { auto_load: false })
+      result_value.gsub!("{{#{query}}}", result)
+    end
+    result_value.to_json
+  end
+
+  def references
+    all_refs = []
+    all_queries.each do |query|
```
Comment (brad-larson) [inline lib/geoengineer/gps/yaml_tag.rb]: just do a `map` here

File: spec/gps/yaml_tag_spec.rb [hunk 3]
```diff
@@ -6,6 +6,28 @@
   let(:nodes) { [n0] }
   let(:constants) { GeoEngineer::GPS::Constants.new({ "e": { "here": "hello" } }) }
 
+  context '!sub' do
```
Comment (brad-larson) [inline spec/gps/yaml_tag_spec.rb]: can you make sure it `to_yaml`'s correctly?

File: lib/geoengineer/gps/yaml_tag.rb (hunk 2 above)
Comment (grahamjenson) [reply lib/geoengineer/gps/yaml_tag.rb]: tru

**Pull request merged**